bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/BookieDeferredSyncTest.java

```diff
@@ -0,0 +1,197 @@
+                .withRecovery(true)
+                .withPassword(new byte[0])
+                .execute());) {
+
+            try (LedgerEntries entries = readLh.read(0, n - 1)) {
+                for (int i = 0; i < n; i++) {
+                    org.apache.bookkeeper.client.api.LedgerEntry entry = entries.getEntry(i);
+                    assertEquals("entry-" + i, new String(entry.getEntryBytes()));
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testCloseNoForce() throws Exception {
+        WriteHandle lh = result(bkc.newCreateLedgerOp()
+                .withEnsembleSize(1)
+                .withWriteQuorumSize(1)
+                .withAckQuorumSize(1)
+                .withWriteFlags(WriteFlag.DEFERRED_SYNC)
+                .withDigestType(org.apache.bookkeeper.client.api.DigestType.CRC32C)
+                .withPassword(new byte[0])
+                .execute());
+
+        int n = 10;
+
+        long ledgerId = lh.getId();
+
+        for (int i = 0; i < n; i++) {
+            lh.append(("entry-" + i).getBytes());
+        }
+
+        // this will close metadata, writing LastAddConfirmed = -1
+        assertEquals(-1, lh.getLastAddConfirmed());
+        lh.close();
+
+        restartBookies();
+
+        try (ReadHandle readLh = result(bkc.newOpenLedgerOp()
+                .withLedgerId(ledgerId)
+                .withRecovery(true)
+                .withPassword(new byte[0])
+                .execute());) {
+            assertEquals(-1, readLh.getLastAddConfirmed());
+        }
+
+    }
+
+    @Test
+    public void testCloseWithForce() throws Exception {
+        WriteHandle lh = result(bkc.newCreateLedgerOp()
+                .withEnsembleSize(1)
+                .withWriteQuorumSize(1)
+                .withAckQuorumSize(1)
+                .withWriteFlags(WriteFlag.DEFERRED_SYNC)
+                .withDigestType(org.apache.bookkeeper.client.api.DigestType.CRC32C)
+                .withPassword(new byte[0])
+                .execute());
+
+        int n = 10;
+
+        long ledgerId = lh.getId();
+
+        for (int i = 0; i < n; i++) {
+            lh.append(("entry-" + i).getBytes());
+        }
+
+        result(lh.force());
+        assertEquals(n - 1, lh.getLastAddConfirmed());
+
+        lh.close();
+
+        restartBookies();
+
+        try (ReadHandle readLh = result(bkc.newOpenLedgerOp()
+                .withLedgerId(ledgerId)
+                .withRecovery(true)
+                .withPassword(new byte[0])
+                .execute());) {
+
+            try (LedgerEntries entries = readLh.read(0, n - 1)) {
+                for (int i = 0; i < n; i++) {
+                    org.apache.bookkeeper.client.api.LedgerEntry entry = entries.getEntry(i);
+                    assertEquals("entry-" + i, new String(entry.getEntryBytes()));
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testForceRoundTripWithDeferredSync() throws Exception {
```

Comment:
`RoundTrip`, what does that mean?

Reply:
renamed and consolidated in only one method.
I meant the full trip from client to bookie and back again to the client, that is testing the full network stack for force RPC.